### ISSUE TITLE
Search within Explore is restricted to facets pre-selected for indexing

### ISSUE THREAD

**Issue by gabsie** [DataBiosphere/data-browser#330]
Hi, this one might also be for later.

I wonder why the Search by default has the projects dropdown. 
I think users should be free to search for whatever term, and not be constrained by what's available, and specifically not by projects as it's limiting their options and it's not possible to know if they are missing out on other data (also affecting efficiency and visibility principles) 

Maybe we should think and address in more detail over what the most important information to search for. I don't mind thinking how to best approach and providing useful feedback.

All the best,
Gabby
![screen shot 2018-10-30 at 14 50 16](https://user-images.githubusercontent.com/19713125/47726958-7057d200-dc53-11e8-851f-d9662707b03c.png)

┆Issue Number: HCAB-291

**Comment by theathorn** [comment]
The Projects list is just the first facet in the search drop down but all the others ("Genus Species", "Biological Sex", "Organ", etc.) are available if you scroll down. Also starting to type a facet name (e.g. "pan" for "pancreas") will limit the list to just the matching items.

Gabby, did you mean something else or can we close this ticket?

**Comment by NoopDog** [comment]
Hi Gabby, we could message this better but the search is over the facet terms. Currently projects are terms. The facet search gives a way to quickly discover all the facets and search over them. We can remove specific  facets if they clutter up the search and also work on the messaging besides just search. 

We can do "Search Facets" or similar.

**Comment by malloryfreeberg** [comment]
Why is there not the ability to search any string a user wants and return all data that have that string somewhere in the metadata? It's not really intuitive that the "search bar" @gabsie originally posted about is not really a search bar, but a concatenate drop down list of pre-selected facets.

In other words, are users only allowed to search exact matches to pre-selected facet values? I made a related issue #330 about searching the string "10x" and actually getting "brain" data.

**Comment by NoopDog** [comment]
Hi @malloryfreeberg  regarding:

>Why is there not the ability to search any string a user wants and return all data that have that string somewhere in the metadata?

The search is currently a match against the existing terms and not a "google like search with synonyms and stemming etc over the facet terms." We would love to implement that its on our backlog, but just not implemented yet. 

Currently the search bar allows for discovery of the facet set by autocomplete filtering over the facets and by showing the full set of facets if there is no match. This at least lets the user see what facets are there. 

That said it does not look to me like the filtering is working as well as it could. I thought we had a partial match not just a "starts with match" on the terms.

so "10x" is not matching anything nor does "10" or "x" but "br" limits to "brain" and "broad".

We will look into adjusting the filter for starters to user contains rather than starts with and take a look at prioritizing other search improvements.

Some of the feedback from the first round of usability testing was that very few people entered thorough the search bar. We are considering messaging to improve that as well.

Regarding
> the search bar is.. a concatenate drop down list of pre-selected facets.

I should mention that he term list list is live and matches the data. Not sure if that was apparent. The terms in the search bar are the exact same ones as the terms under the facets in the drop downs, you just don't have to hunt and peck under the drop downs to find the term you are looking for. The terms are different in the different environments because the data is different.

Finally, we have in flight updates to not automatically select the first term when there is no match. This is what is happening in your case. Brain is the first term and you entering 10x, there is no match, but we highlight the first term any how which is "brain", then you hit enter as we show you the brain. In the next day or two we will not select the first item when there is no match.

In summary:

1. We will look at modifying the autocomplete comparison to do something smarter than "starts with".
1. We will not automatically select the first item when there is no good match. 
1. We will look at a fuller search against the other backlog  items in near future.

Some if the above is also duped by this ticket https://app.zenhub.com/workspace/o/humancellatlas/data-portal/issues/285

Cheers,
Dave

**Comment by malloryfreeberg** [comment]
Thanks for all the clarifications, @NoopDog !

> > the search bar is.. a concatenate drop down list of pre-selected facets.

> I should mention that he term list list is live and matches the data. 

I understand that the list is live and is populated by actual values in the metadata. I should clarify that I mean it appears the list is restricted to matching values only in *fields that have been chosen as facets*, not all possible metadata fields.

So, for example, it doesn't appear the `project_description` field is a facet, so I can't search within that field at all for my string of interest. Is this true?

**Comment by NoopDog** [comment]
You are correct @malloryfreeberg.  If its not one of the terms in one of the facets in the "mini schema" it wont show up. This is not a search over the data. This is a search over the exposed facets and their terms. 

Perhaps at the F2F we can get with @hannes-ucsc and review what actually is indexed and available for free text search. The indexes are currently geared toward serving the facet search. The "search" is geared to help with facet and term discovery and to easily find and select a term if you know its there. 

Very happy to discuss what next steps on the the search might be.

Cheers,
D

**Comment by malloryfreeberg** [comment]
I imagine @adriennes and @gabsie will have a lot of perspective on how users might "search" in the Browser, so happy to have a discussion at the F2F :)

**Comment by theathorn** [comment]
See https://app.terra.bio/ for an example of a browser which allows the user to select any facet of interest and configure into the UI.

**Comment by NoopDog** [comment]
Obsolete
